commonmark/src/main/java/org/commonmark/parser/SourceLine.java

```diff
@@ -44,4 +66,41 @@ public SourceLine substring(int beginIndex, int endIndex) {
         }
         return SourceLine.of(newContent, newSourceSpan);
     }
+    
+ // Many lines are passed around as raw strings, with an index value indicating
+    //    where the literal string begins. Calling this method returns a SourceLine containing
+    //    only the literal portion of the line (if it is a subset of the line) or the line
+    //    in full (if they are the same).
+    public SourceLine getLiteralLine() {
```

Comment:
Yeah, it may be wise to standardize this somehow.

I noticed that anytime you just wanted to just "get the text", you had a `getLiteral` method (see `HtmlBlock`, `Code`, `Text`, etc.). This literal value is used by HTML rendering and is rarely unprocessed, as it's always at least post-block-processing and sometimes post-inline as well. I chose to use the term "raw" whenever lines needed to be as unprocessed as possible. That same mindset slipped into a couple other places, so `getLiteralLine` here as well as the concept of a "literal index" (an offset which, if you ignore the contents prior to that index, you get what the HTML renderer would expect).

I'm open to suggestion on how that might be communicated more clearly. The line you see here does have a specific purpose, but I would want it to be a clear purpose to everyone who isn't me, too!